### DISCUSSION TITLE
Error when executing the script via bash

### DISCUSSION THREAD

**Discussion by CMBoii** [Taxel/PlexTraktSync#649] [Support] [answered]
### The problem

I'm currently using the version 0.17.0. Script installed with git clonte & pip (not pipx/pipenv)

It seems that if I run the script by using bash path/to/script.sh it will throw the follwoing error which didn't happen in previous versions of the script. Unfortunately, I don't remember which version worked properly. 

### Error trace

```text
/volume1/docker/trakt/PlexTraktSync/plextraktsync.sh: line 2: set: -: invalid option
set: usage: set [-abefhkmnptuvxBCHP] [-o option-name] [--] [arg ...]
/volume1/docker/trakt/PlexTraktSync/plextraktsync.sh: line 3: $'\r': command not found
/volume1/docker/trakt/PlexTraktSync/plextraktsync.sh: line 5: cd: $'/volume1/docker/trakt/PlexTraktSync\r\r': No such file or directory
/volume1/docker/trakt/PlexTraktSync/plextraktsync.sh: line 6: $'\r': command not found
/bin/python3: No module named plextraktsync
```

### Expected behavior

Run script. The script works by using python3 -m plextraktsync. However, as I'm a synology user, if you want to use task scheduler you need to set the path therefore you can't use that command so you need to use the bash one described above. 

### Install method

git clone

### Version

0.17.0

### Python Version

3.8.8

### Operating System and Version

Synology DSM 7

**Comment by glensc** [comment, marked as answer]
> `$'\r': command not found`

this means that your files are CRLF formatted (DOS, Windows), but the bash expects them to be LF formatted (Unix, Linux).

You can use the `dos2unix` tool to convert the line endings.

> ```
> line 2: set: -: invalid option
> set: usage: set [-abefhkmnptuvxBCHP] [-o option-name] [--] [arg ...]
> ```

this error is very odd. what is the version of your bash? also what does `/bin/sh` point to?

**Reply by CMBoii** [reply]
v4.4.23.  Where does it say /bin/sh?

I managed to install the script with pipx the only problem is that I can't use the task scheduler (kind of cron job) since it wouldn't be a able to run the script from there.

**Reply by CMBoii** [reply]
It seems that manually downloading the source code instead of cloning it fixed it. I think it might be related to the CRLF format you mentioned before. 

Thanks for your help.

**Comment by glensc** [comment]
`/bin/sh` is the shebang of the script you were using:
- https://github.com/Taxel/PlexTraktSync/blob/0.17.0/plextraktsync.sh#L1

showing what is your `/bin/sh` with `ls -l`:

```
➔ LC_ALL=C ls -l /bin/sh
-rwxr-xr-x 1 root wheel 121008 Oct 18 06:30 /bin/sh*
```

it might be a symlink to `/bin/bash`

**Reply by glensc** [reply]
This should force proper line ending when using git clone:
- https://github.com/Taxel/PlexTraktSync/pull/651
